docs/docs/gitsync.md

```diff
@@ -0,0 +1,229 @@
+
+<div style={{textAlign: 'center'}}>
+
+<img className="screenshot-full" src="/img/gitsync/github2.png" alt="GitHub" />
+
+</div>
+
+### Step 3: Configure the GitSync feature on ToolJet
+
+Go to the **Workspace settings**, and click on the **Configure git** tab.
+
+<div style={{textAlign: 'center'}}>
+
+<img className="screenshot-full" src="/img/gitsync/gitsync.png" alt="GitSync" />
+
+</div>
+<br/>
+
+Enter the **SSH URL** of the repository (obtained in Step 2) in the **Git Repository URL** field. Click on the **Generate SSH Key** button, and copy the SSH key that is generated. The SSH key is used to authenticate ToolJet with the repository.
+
+<div style={{textAlign: 'center'}}>
+
+<img className="screenshot-full" src="/img/gitsync/git2.png" alt="GitSync" />
+
+</div>
+
+### Step 4: Deploy the SSH key to GitHub repository
+
+Go to the **Settings** tab of the GitHub repository that you created in Step 1, and click on the **Deploy keys** tab. Click on the **Add deploy key** button. 
+
+<div style={{textAlign: 'center'}}>
+
+<img className="screenshot-full" src="/img/gitsync/github3.png" alt="GitHub" />
+
+</div>
+
+Enter a title for the SSH key in the **Title** field. Paste the SSH key that you copied in Step 3 in the **Key** field. Make sure that the **Allow write access** checkbox is checked, especially when configuring the GitSync feature to [push changes to Git](#pushing-changes-to-git-repo). However, it is not mandatory to check this option when setting up the GitSync feature for [pulling changes from Git](#pulling-changes-from-git-repo). Finally, click on the **Add key** button.
+
+<div style={{textAlign: 'center'}}>
+
+<img className="screenshot-full" src="/img/gitsync/github4.png" alt="GitHub" />
+
+</div>
+
+### Step 5: Finish the GitSync configuration on ToolJet
+
+Go back to the **Configure git** tab on ToolJet, and click on the **Finalize setup** button. If the SSH key is configured correctly, you will see a success message.
+
+<div style={{textAlign: 'center'}}>
+
+<img className="screenshot-full" src="/img/gitsync/git5.png" alt="GitSync" />
+
+</div>
+
+## Enable/Disable GitSync
+
+To enable or disable the GitSync feature, go to the **Configure git** tab on the **Workspace settings** page, and toggle on/off the **Connect** switch. This is only available if the GitSync feature is configured.
+
+<div style={{textAlign: 'center'}}>
+
+<img className="screenshot-full" src="/img/gitsync/connect.png" alt="GitSync" />
+
+</div>
+
+## Delete GitSync configuration
+
+To delete the GitSync configuration, go to the **Configure git** tab on the **Workspace settings** page, and click on the **Delete configuration** button. This will delete the SSH key from the GitHub repository and the GitSync feature will be disabled.
+
```

Comment:
The last statement is incorrect, we are just deleting data in db. We should define the delete behaviour properly. Can you please take a look once?